api/src/test/java/org/openmrs/module/fhirExtension/translators/DiagnosticReportObsTranslatorHelperTest.java

```diff
@@ -0,0 +1,216 @@
+package org.openmrs.module.fhirExtension.translators;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openmrs.*;
+import org.openmrs.api.ConceptService;
+import org.openmrs.module.fhirExtension.domain.observation.LabResult;
+import org.openmrs.module.fhirExtension.translators.impl.DiagnosticReportObsTranslatorHelper;
+
+import java.util.*;
+
+import static java.util.Locale.ENGLISH;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DiagnosticReportObsTranslatorHelperTest {
+	
+	private static final String REPORT_URL = "/100/uploadReport.pdf";
+	
+	private static final String REPORT_NAME = "bloodTest.pdf";
+	
+	private static final String LAB_TEST_NOTES = "Report is normal";
+	
+	private static final int PATIENT_ID = 12345;
+	
+	@Mock
+	private ConceptService conceptService;
+	
+	@InjectMocks
+	private final DiagnosticReportObsTranslatorHelper diagnosticReportObsResultTranslatorHelper = new DiagnosticReportObsTranslatorHelper();
+	
+	@Test
+    public void shouldTranslateToObsWhenTestReportIsUploadedWithLabNotesForSingleTestWithoutAnyOrder() {
```

Comment:
this name is hard... you can use a separator to help
`givenSingleTest_WhenTestReportIsUploadedWithNotes_ShouldTranslateToObs`

Reply:
Also does the requirements differ for single vs multiple test? ordered or non ordered test?

Reply:
> this name is hard... you can use a separator to help `givenSingleTest_WhenTestReportIsUploadedWithNotes_ShouldTranslateToObs`

Test name updated.